src/config_file.c

```diff
@@ -35,6 +35,7 @@ gboolean load_config(const gchar *filename, RaucConfig **config, GError **error)
 	gsize group_count;
 	GList *slotlist = NULL;
 	GHashTable *slots = NULL;
+	GHashTable *bootnames = NULL;
```

Comment:
If we use autoptr here, we could eliminate the freeing below

```suggestion
	g_autoptr(GHashTable) bootnames = NULL;
```

Reply:
Okay, I'll have to read about g_autoptr. It's been quite a few years since I wrote GLib code and back then it didn't have autoptr.

I suppose the autoptr might be useful for fixing the other leaks in the error paths too, so I can try that in a separate PR.